### PR TITLE
VIMC-3229 make burden estimate downloads work for all groups

### PR DESCRIPTION
**Pull request by hillalex**
Accompanying db migration - https://github.com/vimc/montagu-db/pull/141. Once that's merged, the db version here should be pointed back to master.

The Travis builds are failing because they use the master db image, which doesn't yet contain the crosstab function (and they have to use master because we don't push branch tags to the public docker hub.)

### CODE REVIEW

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt [hunk 1]
```diff
@@ -221,58 +219,80 @@ class JooqBurdenEstimateRepository(
...
 
-    override fun getBurdenEstimateOutcomesSequence(groupId: String, touchstoneVersionId: String, scenarioId: String, burdenEstimateSetId: Int)
-            : Sequence<BurdenEstimateOutcome>
+    override fun getBurdenEstimateOutcomesSequence(burdenEstimateSetId: Int,
+                                                   outcomes: List<Pair<Short, String>>,
+                                                   disease: String)
+            : Sequence<BurdenEstimate>
     {
-        //check that the burden estimate set exists in the group etc
-        val set = getBurdenEstimateSet(groupId, touchstoneVersionId, scenarioId, burdenEstimateSetId)
```
Comment (hillalex) [inline src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt]: This logic, and the stochastic check, moved to the logic layer

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt [hunk 2]
```diff
@@ -221,58 +219,80 @@ class JooqBurdenEstimateRepository(
...
+                .from(BURDEN_OUTCOME)
+                .where(BURDEN_OUTCOME.CODE.eq("cohort_size"))
+                .fetchAnyInto(Short::class.java)
+
+        return countryLookup.keys.asSequence().map { k ->
+            val cursor = dsl.fetchLazy(getBurdenEstimateQueryForCountry(burdenEstimateSetId, outcomes, cohortSize, k))
+            generateSequence {
+                cursor.fetchNext()?.map { record ->
+                    val country = countryLookup.getValue(record.get("country", Short::class.java))
+                    val outcomeList = outcomes.filter { it.second != "cohort_size" }
```
Comment (EmmaLRussell) [inline src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt]: Cohort size is treated as an outcome for burden estimate sets? Could do this filter outside the lambda I guess.

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt [hunk 3]
```diff
@@ -221,58 +219,80 @@ class JooqBurdenEstimateRepository(
...
+    }
 
+    private fun getBurdenEstimateQueryForCountry(setId: Int,
+                                                 outcomes: List<Pair<Short, String>>,
+                                                 cohortSize: Short,
+                                                 country: Short): String
+    {
+        val outcomeIdQuery = "select ${outcomes.map { it.first }.joinToString(" union select ")} union select $cohortSize order by 1"
+        val expectedOutcomeCodes = "${outcomes.joinToString(" real,") { it.second }} real, cohort_size real"
+        val compoundRowKey = "year || '':''|| age || '':'' || country"
```
Comment (EmmaLRussell) [inline src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt]: Do you still need country in your compound key now you're only fetching results for one country??

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt (hunk 2 above)
Comment (hillalex) [reply src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt]: (yes cohort_size is an outcome)

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt (hunk 3 above)
Comment (hillalex) [reply src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt]: Oh nice, no I don't think we do

File: src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt (hunk 2 above)
Comment (hillalex) [reply src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt]: Oh, although actually cohort_size shouldn't be in the expectations table, so this isn't needed here

**Comment by hillalex** [comment]
Travis build is error-ing because it's failing to install codecov. I presume this is an intermittent issue that will go away.
```
70.95s$ npm install codecov -g
npm ERR! code E429
npm ERR! 429 Too Many Requests - GET https://registry.npmjs.org/codecov
```

**Pull request merged**